addOns/graphql/src/main/java/org/zaproxy/addon/graphql/ExtensionGraphQl.java

```diff
@@ -49,6 +54,7 @@
     private ParseFilter graphQlParseFilter;
     private GraphQlOptionsPanel graphQlOptionsPanel;
     private GraphQlParam param;
+    private List<ParserThread> parserThreads = new ArrayList<>();
```

Comment:
This should be synchronised (e.g. wrap with `Collections.synchronizedList`), to not add while looping (which would lead to a `ConcurrentModificationException`).